In [1]:
import pandas as pd
import import_ipynb
import h1_ENB_occ as h1 
import h2_ENB_gen as h2 

importing Jupyter notebook from h1_ENB_occ.ipynb
importing Jupyter notebook from c1_extract_paragraphe_issue.ipynb
Issue  34
importing Jupyter notebook from c2_extract_sentence_issue.ipynb
importing Jupyter notebook from g1_generate_dictionary.ipynb
importing Jupyter notebook from h2_ENB_gen.ipynb
[nltk_data] Downloading package punkt to
[nltk_data]     /Users/tatianacogne/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /Users/tatianacogne/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!


In [2]:
def extract_list_issue():
     with open("Text/ENB_relationships.txt") as infile:
        k=0
        num = []
        for line in infile:
            x = line.replace('"','').split('\t')[:26]

            if(k > 0 and int(x[23]) < 300  and 250 < int(x[23])):

                num.append(int(x[23]))
            k +=1
        return list(set(num))

In [3]:
list_issue_num = extract_list_issue()

In [4]:
def extract_gen():
    frames1 = []
    frames2 = []
    for issue_number in list_issue_num:
        df1 = h1.extract_relationships_count(issue_number)
        df2 = h2.interactions(issue_number)
        frames1.append(df1)
        frames2.append(df2) 
    return pd.concat(frames1), pd.concat(frames2)


In [5]:
df_original, df_generated = extract_gen()

Issue  256
Issue  257
Issue  258
SIDS
SIDS
SIDS
Issue  259
SIDS
Issue  261
Issue  263
Issue  264
Issue  265
SIDS
SIDS
Issue  266
SIDS
Issue  267
Issue  268
Issue  269
Issue  281
Issue  282
Issue  283
Issue  284
Issue  285
SIDS
Issue  286
Issue  287
Issue  288
SIDS
SIDS
Issue  289
SIDS
Issue  290
SIDS
SIDS
SIDS
Issue  297
Issue  298
Issue  299
Issue  251
SIDS
SIDS
SIDS
Issue  252
LATIN AMERICA
SIDS
SIDS
Issue  253
SIDS
Issue  254
SIDS
Issue  255


In [112]:
df_generated

type  issue  id_ca  id_cb  behalf  support  agreement  opposition  \
47  generated    251      0     15       0        1          0           0   
49  generated    251      0     22       0        0          1           0   
46  generated    251      0     49       0        0          1           0   
48  generated    251      0     77       0        0          1           0   
10  generated    252      0    128       0        0          1           0   
..        ...    ...    ...    ...     ...      ...        ...         ...   
6   generated    252   9999    164       0        0          1           0   
65  generated    265   9999    185       0        0          1           0   
19  generated    288   9999    185       0        0          1           0   
15  generated    285   9999    216       0        0          1           0   
4   generated    252   9999    220       0        0          1           0   

    criticism  cooperation  
47          0            1  
49          0            1  
46          0            1  
48          0            1  
10          0            1  
..        ...          ...  
6           0            1  
65          0            1  
19          0            1  
15          0            1  
4           0            1  

[1957 rows x 10 columns]

In [113]:
df_original.sort_values(by=['id_ca','id_cb'], inplace=True)
df_generated.sort_values(by=['id_ca','id_cb'], inplace=True)

In [114]:
dd1 = df_original.drop(['issue','type','behalf','support','agreement','opposition','criticism'],axis=1).groupby(['id_ca','id_cb']).count()
dd2 = df_generated.drop(['issue','type','behalf','support','agreement','opposition','criticism'],axis=1).groupby(['id_ca','id_cb']).count()

In [115]:
dd1

cooperation
id_ca id_cb             
0     15               1
      22               1
      49               1
      77               1
      103              1
...                  ...
222   171              1
      183              2
      207              2
      216              2
      219              1

[944 rows x 1 columns]

In [24]:
dd2

cooperation
id_ca id_cb             
0     15               1
      22               1
      49               1
      77               1
      128              1
...                  ...
9999  143              2
      164              2
      185              2
      216              1
      220              1

[976 rows x 1 columns]

In [116]:
df = pd.concat([dd1,dd2])
df.sort_values(by=['id_ca','id_cb'], inplace=True)

df

cooperation
id_ca id_cb             
0     15               1
      15               1
      22               1
      22               1
      49               1
...                  ...
9999  143              2
      164              2
      185              2
      216              1
      220              1

[1920 rows x 1 columns]

In [122]:
#Union
dd1 = df_original.drop(['issue','type','behalf','support','agreement','opposition','criticism'],axis=1).groupby(['id_ca','id_cb']).count()
dd2 = df_generated.drop(['issue','type','behalf','support','agreement','opposition','criticism'],axis=1).groupby(['id_ca','id_cb']).count()
dd_union = pd.concat([dd1,dd2]).sort_values(['id_ca','id_cb'])


cooperation
id_ca id_cb             
0     15               1
      15               1
      22               1
      22               1
      49               1
...                  ...
9999  143              2
      164              2
      185              2
      216              1
      220              1

[1920 rows x 1 columns]

missing id dict

problems

' Malaysia and Ghana, both speaking for the G-77/CHINA, suggested some additions.'Issue  283'

In [ ]:
Original : 944 rows × 1 columns
Generatec :976 rows × 1 columns